Filter/ModelFilter.php

```diff
@@ -100,9 +100,11 @@ protected function handleMultiple(ProxyQueryInterface $queryBuilder, $alias, $da
 
             $or->add($queryBuilder->expr()->notIn($alias, ':'.$parameterName));
 
-            $or->add($queryBuilder->expr()->isNull(
-                sprintf('IDENTITY(%s.%s)', $this->getParentAlias($queryBuilder, $alias), $this->getFieldName())
-            ));
+            if (ClassMetadataInfo::MANY_TO_MANY !== $this->getOption('mapping_type')) {
```

Comment:
I personally use the filter with this configuration.

```php
protected function configureDatagridFilters(DatagridMapper $datagridMapper)
    {
        $datagridMapper
            ->add('tags', ModelFilter::class, [
                'mapping_type' => ClassMetadataInfo::MANY_TO_MANY,
            ], null, [
                'class' => ArticleTag::class,
                'choice_label' => function (ArticleTag $value, $key, $index) {
                    return $value->getLabel();
                }
            ])
        ;
    }
```

I'm assigning the value `ClassMetadataInfo::MANY_TO_MANY` to the option.
I don't understand your previous point. Is it just about removing `!==` ?